documentation/specs/Pill.md

```diff
@@ -0,0 +1,156 @@
+# `Pill` Component Specification
+
+## Overview
+
+A `Pill` is a compact element that displays contextual text typically representing selections or options; they can be dismissed as well as accompanied by an icon or image.
+
+### Use Cases
+
+- Display selections or options made by the user or system
+
+### Features
+
+- Supports icons and images
+- Supports dismissal
+
+### Risks and Challenges
+
+- Ensuring ease of use
+- Handling invalid prop configurations
+
+### Prior Art
+
+Not a 1 to 1, but similar:
+
+- [MUI `<Chip />`](https://mui.com/material-ui/react-chip/)
+- [Polaris `<Tag />`](https://polaris.shopify.com/components/selection-and-input/tag?example=tag-removable)
+- [React Aria `<TagGroup />`](https://react-spectrum.adobe.com/react-aria/TagGroup.html)
+
+---
+
+## Design
+
+A `Pill` is a fairly straightforward component with no external dependencies and it does not require the use of any React Aria primitives. For icons and dismissal support, this component will follow existing patterns in Easy UI.
+
+### API
+
+```ts
+export type PillProps = {
+  /** Text label */
+  children: ReactNode;
+  /** Left aligned icon */
+  icon?: IconSymbol;
+  /** Left aligned image */
+  image?: ReactNode;
```

Comment:
**thought**: up to you on what you think is best, but the `Icon` component i believe is designed to accept images as symbols which would allow us to use the constraints (size, color, etc) built into the `Icon` component with images, and allow the component to use a single prop.

here's the gist of how it could work:

```
type PillProps = {
  iconSymbol?: IconSymbol;
};

function Component({ carrier }) {
  // This would have to be an element passed in as a symbol
  const CarrierImage = () => <Image src={`/${carrier}.png`} />;
  return (
    <Pill iconSymbol={CarrierImage}>
      First Last #12345
    </Pill>
  );
}

<span className={className}>
  {iconSymbol && <Icon size="xs" symbol={iconSymbol} />}
  <Text>{children}</Text>
  <UnstyledButton>
    <Icon size="xs" symbol={CloseIcon} />
  </UnstyledButton>
</span>
```